song-server/src/main/java/bio/overture/song/server/service/StudyService.java

```diff
@@ -60,16 +61,23 @@ public boolean isStudyExist(String studyId){
   }
 
   @Transactional
-  public String saveStudy(Study study) {
-    val id = study.getStudyId();
+  public String validatedSaveStudy(Study actualStudy, String expectedStudyId) {
+    val id = actualStudy.getStudyId();
+    checkServer(id.equals(expectedStudyId), getClass(), MISMATCHING_STUDY_IDS,
+            "The studyId in the URL '%s' should match the studyId '%s' in the payload",
+            id, actualStudy.getStudyId());
```

Comment:
on second thought, move this checkServer call to the controller, and rename the method to just saveStudy. the controller should be responsible for the validation in that context. Also change the tests to use the StudyController, and just make the call there to validate the controller errors out properly.